src/js_parser.zig

```diff
@@ -17355,9 +17356,151 @@ fn NewParser_(
         ) []Stmt {
             switch (stmtorexpr) {
                 .stmt => |stmt| {
-                    var stmts = p.allocator.alloc(Stmt, 1) catch unreachable;
-                    stmts[0] = stmt;
-                    return stmts;
+                    var class = stmt.data.s_class.class;
+                    var ctor: ?*E.Function = null;
+
+                    var hasDecorators: bool = class.ts_decorators.len > 0;
+                    for (class.properties) |prop| {
+                        if (prop.ts_decorators.len > 0) {
+                            hasDecorators = true;
+                            break;
+                        }
+                    }
+
+                    if (!hasDecorators) {
+                        var stmts = p.allocator.alloc(Stmt, 1) catch unreachable;
+                        stmts[0] = stmt;
+                        return stmts;
+                    }
+
+                    var stmts = ListManaged(Stmt).init(p.allocator);
+
+                    var instanceMembers = ListManaged(Stmt).init(p.allocator);
+
+                    var instanceDecorators = ListManaged(Stmt).init(p.allocator);
+                    var staticDecorators = ListManaged(Stmt).init(p.allocator);
+
+                    var updatedClassProps = ListManaged(G.Property).init(p.allocator);
+
+                    for (class.properties) |prop| {
+
+                        // handle decorators
+                        if (prop.ts_decorators.len > 0) {
+                            var loc = prop.key.?.loc;
```

Comment:
use `const` instead of `var` by default

`const` produces faster code than `var`

Reply:
👍